### ISSUE TITLE
How to get unwrapped error from retry?

### ISSUE THREAD

**Issue by ITChelik** [avast/retry-go#65]
In retry func, if I have error, i return `errors.Wrapf(someErr, "someWrap text", some wrap values)`.
And retry func return me this type for error:
![image](https://user-images.githubusercontent.com/107698898/185943036-dd076e94-8622-431c-9e51-dc8a92edd5f8.png)

But I need extract `someErr`, from returned error.
Usually I do this by this code:
```
for errors.Unwrap(err) != nil {
		err = errors.Unwrap(err)
}
```

But with retry this method doesn't work.

Full code:
```
err = retry.Do(
		func() error {
			defer func() { sentCount++ }()
			responseBody, statusCode, err = h.Request(ctx, args.RequestParams)
			if err != nil {
				return errors.Wrapf(err, "fhttp.RequestWithRetry(args: %s)", utils.GetStructJSON(args)) // if last retry this need to unwrap err
			}

			if statusCode != args.ExpectedStatusCode {
				if statusCode == fasthttp.StatusUnauthorized {
					return ErrStatusUnauthorized
				}
				return errors.Wrapf(
					ErrNotExpectedStatusCode,
					"fhttp.RequestWithRetry, expeced %d, got %d statusCode; requestArgs: %s, responseBody: %s",
					args.ExpectedStatusCode,
					statusCode,
					utils.GetStructJSON(args),
					responseBody,
				)  // if last retry in this part of code need possibility unwrap to ErrNotExpectedStatusCode
			}
			return nil
		},
		retry.Delay(args.RetryDelay*time.Millisecond),
		retry.RetryIf(func(err error) bool {
			if ((errors.Is(err, fasthttp.ErrTimeout) ||
				errors.Is(err, fasthttp.ErrDialTimeout) ||
				errors.Is(err, fasthttp.ErrTLSHandshakeTimeout)) && sentCount >= 2) ||
				errors.Is(err, ErrStatusUnauthorized) {
				return false
			}
			return true
		}),
		retry.Attempts(args.RetryAttempts),
	)
        for errors.Unwrap(err) != nil { // do nothing 
		err = errors.Unwrap(err)
	}
```

**Comment by JaSei** [comment]
it should be solved by #70

**Comment by JaSei** [comment]
solved in version 4.1.0
